proposals/nnnn-app-services.md

```diff
@@ -0,0 +1,629 @@
+		<description> true if successful; false, if failed </description>
+	</param>
+       
+	<param name="resultCode" type="Result" platform="documentation" mandatory="true">
+		<description>See Result</description>
+		<element name="SUCCESS"/>
+		<element name="INVALID_DATA"/>
+		<element name="OUT_OF_MEMORY"/>
+		<element name="TOO_MANY_PENDING_REQUESTS"/>
+		<element name="APPLICATION_NOT_REGISTERED"/>
+		<element name="GENERIC_ERROR"/>
+	</param>
+	
+	<param name="info" type="String" maxlength="1000" mandatory="false" platform="documentation">
+		<description>Provides additional human readable info regarding the result.</description>
+	</param>
+	
+	<param name="offset" type="Integer" minvalue="0" maxvalue="2000000000" mandatory="false">
+		<description>Optional offset in bytes for resuming partial data chunks</description>
+	</param>
+	
+	<param name="length" type="Integer" minvalue="0" maxvalue="2000000000" mandatory="false">
+		<description>Optional length in bytes for resuming partial data chunks if offset is set to 0, then length is the total length of the file to be downloaded</description>
+	</param>
+
+	
+</function>
+```
+
+#### App Service Actions - `PerformAppServiceInteraction`
+
+Talk about how this works here.
+
+```xml
+<function name="PerformAppServiceInteraction" functionID="PerformAppServiceInteractionID" messagetype="request">
+
+	<param name="serviceUri" type="String"  mandatory="true"/>
+	<param name="appServiceId" type="String" mandatory="false"/>
+	<param name="requestServiceActive" type="Boolean" mandatory="false"/>
+
+		
+</function>
+
+<function name="PerformAppServiceInteraction" functionID="PerformAppServiceInteractionID" messagetype="response">
+
+	<param name="success" type="Boolean" platform="documentation" mandatory="true">
+		<description> true if successful; false, if failed </description>
+	</param>
+       
+	<param name="resultCode" type="Result" platform="documentation" mandatory="true">
+		<description>See Result</description>
+		<element name="SUCCESS"/>
+		<element name="INVALID_DATA"/>
+		<element name="OUT_OF_MEMORY"/>
+		<element name="TOO_MANY_PENDING_REQUESTS"/>
+		<element name="APPLICATION_NOT_REGISTERED"/>
+		<element name="GENERIC_ERROR"/>
+	</param>
+	
+	<param name="info" type="String" maxlength="1000" mandatory="false" platform="documentation">
+		<description>Provides additional human readable info regarding the result.</description>
+	</param>
+
+	
+</function>
+```
+
+### Core
+
+Core will have to manage which services are currently published and which of those are considered active. If a user wishes to select an app service over an onboard service there should be an easy way to perform this action on the IVI system.
+
+
+##### Activating Service Providers
+
+Enabling services will be a major UX concern for Core and HMI portions. There should exists a set of guidelines to ensure the consumer will understand whats happening and how to change service providers.
```

Comment:
Please update second sentence to "A set of guidelines should exist to ensure the consumer will understand what's happening..."